pkg/expiringregistry/expiringregistry.go

```diff
@@ -276,10 +315,9 @@ func (r *registry) getHistogram(metricName string, labels prometheus.Labels, hel
 
 	var histogramVec *prometheus.HistogramVec
 	if vh == nil {
-		metricsCount.WithLabelValues("histogram").Inc()
-		buckets := r.mapper.Defaults.Buckets
-		if mapping.Buckets != nil && len(mapping.Buckets) > 0 {
-			buckets = mapping.Buckets
+		r.metricsCount.WithLabelValues("histogram").Inc()
+		if buckets == nil || len(buckets) == 0 {
+			buckets = r.defaults.Buckets
```

Comment:
hmmmm so this is where the mapping and the expiration got very intertwined. What would the mapper need to return to make it unnecessary to handle defaults here? Would it make sense if the mapper returns a complete `prometheus.HistogramOpts` (and `prometheus.SummaryOpts` if timer type is summary) every time?

Conceptually, I think expiring registry needs to know about _Prometheus_ domain objects (metric names and types, labels and values, help, summary/histogram options) and _time_ (so that it can expire the former). It _should_ not need to care how they came to into being. Consequently, ideally the `expiringregistry` package should not have a dependency on the `mapper` package.